tools/devtool

```diff
@@ -731,6 +734,84 @@ cmd_tag() {
     say "Tag v$version created."
 }
 
+# Check if able to run firecracker.
+# https://github.com/firecracker-microvm/firecracker/blob/master/docs/getting-started.md#prerequisites
+# https://github.com/firecracker-microvm/firecracker/blob/master/docs/prod-host-setup.md
+ensure_kvm_rw () {
+    [[ -c /dev/kvm && -w /dev/kvm && -r /dev/kvm ]] || \
+        say_err "FAILED: user $(whoami) doesn't have permission to" \
+                "access /dev/kvm. You won't be able to run Firecracker"
+}
+check_SMT () {
+    (grep -q "^forceoff$\|^notsupported$" \
+	  /sys/devices/system/cpu/smt/control) ||
+	say_err "FAILED: Hyperthreading ENABLED."
+}
+check_KPTI () {
+    (grep -q "^Mitigation: PTI$" \
+	  /sys/devices/system/cpu/vulnerabilities/meltdown) || \
+	say_err "FAILED: KPTI NOT SUPPORTED"
+}
+check_KSM () {
```

Comment:
KSM doesn't break Firecracker.

Reply:
yes, I should have added as a warning. I have changed the security checks as warnings and point to the  [Production Host Setup](https://github.com/firecracker-microvm/firecracker/blob/master/docs/prod-host-setup.md)

Reply:
@aghecenco , @cneira it might be worth going a bit further here.

While the things in [Production Host Setup](https://github.com/firecracker-microvm/firecracker/blob/master/docs/prod-host-setup.md) don't break Firecracker, they should always be applied to a host running production multi-tenant workloads with Firecracker. And, running production multi-tenant workloads is pretty much the use-case we exclusively optimizing for.

So, perhaps it would make sense to word warnings so as to indicate that the environment is not safe for production. Or there could be two commands to check environments: one for development, and one for production; in this case, the two commands should explicitly state what they are fit for.

Finally, any mention about host production safety should come with a wording like "as far as we can tell ... :)." since while we will do our best here, there's never an absolute guarantee in this space.

Reply:
+1 for two separate checks: development and production.